# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/power/WakeLockJS.h
# Commit: e02e11c3d977
# Full Hash: e02e11c3d97779c71984db55459ebc843d6cb303
# Author: Vincent Hilla <vhilla@mozilla.com>
# Date: 2023-12-05 21:29:44
# Regressor Bug: 1589554
# File Overlap Count: 1
# Description:
#   Bug 1589554 - Part 4: Release/Deny screen-wake-lock for implementation-specific reasons. r=dom-core,edgar
#   
#   Depends on D189510
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D189511
# ==============================================================================

diff -r 202c4f5c642b -r e02e11c3d977 dom/power/WakeLockJS.h
--- a/dom/power/WakeLockJS.h	Tue Dec 05 15:37:06 2023 +0000
+++ b/dom/power/WakeLockJS.h	Tue Dec 05 15:37:06 2023 +0000
@@ -9,10 +9,13 @@
 
 #include "js/TypeDecls.h"
 #include "mozilla/Attributes.h"
+#include "mozilla/HalBatteryInformation.h"
 #include "mozilla/dom/WakeLockBinding.h"
 #include "nsIDOMEventListener.h"
 #include "nsIDocumentActivity.h"
+#include "nsIObserver.h"
 #include "nsCycleCollectionParticipant.h"
+#include "nsWeakReference.h"
 #include "nsWrapperCache.h"
 
 class nsPIDOMWindowInner;
@@ -38,10 +41,14 @@
  */
 class WakeLockJS final : public nsIDOMEventListener,
                          public nsWrapperCache,
-                         public nsIDocumentActivity {
+                         public hal::BatteryObserver,
+                         public nsIDocumentActivity,
+                         public nsIObserver,
+                         public nsSupportsWeakReference {
  public:
   NS_DECL_NSIDOMEVENTLISTENER
   NS_DECL_NSIDOCUMENTACTIVITY
+  NS_DECL_NSIOBSERVER
 
   NS_DECL_CYCLE_COLLECTING_ISUPPORTS
   NS_DECL_CYCLE_COLLECTION_WRAPPERCACHE_CLASS_AMBIGUOUS(WakeLockJS,
@@ -59,6 +66,8 @@
   JSObject* WrapObject(JSContext* aCx,
                        JS::Handle<JSObject*> aGivenProto) override;
 
+  void Notify(const hal::BatteryInformation& aBatteryInfo) override;
+
   already_AddRefed<Promise> Request(WakeLockType aType, ErrorResult& aRv);
 
  private:
@@ -82,8 +91,6 @@
   Result<already_AddRefed<WakeLockSentinel>, RequestError> Obtain(
       WakeLockType aType);
 
-  void UnlockAll(WakeLockType aType);
-
   RefPtr<nsPIDOMWindowInner> mWindow;
 };
 